src/app/src/components/annotations/annotator.tsx

```diff
@@ -520,6 +523,106 @@ export default class Annotator extends Component<
+      this.state.assetList.map(async asset => {
+        if (
+          asset.type === "image" &&
+          this.state.inferenceOptions.bulkAnalysisStatus !== "video"
+        ) {
+          await APIGetInferenceFlask(
+            loadedModelHash,
+            asset.localPath,
+            this.state.inferenceOptions.iou,
+            "json"
+          )
+            .then(response => {
+              if (this.currentAsset.url === asset.url) {
+                this.updateAnnotations(response.data);
+              }
+            })
+            .catch(error => {
+              let message = `Failed to predict image. ${error}`;
+              if (error.response) {
+                message = `${error.response.data.error}: ${error.response.data.message}`;
+              }
+
+              CreateGenericToast(message, Intent.DANGER, 3000);
+            });
+        }
+
+        if (
+          asset.type === "video" &&
+          this.state.inferenceOptions.bulkAnalysisStatus !== "image"
+        ) {
+          await APIGetVideoPrediction(
+            loadedModelHash,
+            asset.localPath,
+            this.state.inferenceOptions.video.frameInterval,
+            this.state.inferenceOptions.iou
+          )
+            .then(response => {
+              if (this.currentAsset.url === asset.url) {
+                const videoElement = this.videoOverlay.getElement();
+                const videoFrameCallback = (
+                  now: DOMHighResTimeStamp,
+                  metadata: VideoFrameMetadata
+                ) => {
+                  const secondsInterval =
+                    this.state.inferenceOptions.video.frameInterval /
+                    response.data.fps;
+                  const quotient = Math.floor(
+                    metadata.mediaTime / secondsInterval
+                  );
+
+                  const key = Math.floor(
+                    quotient * secondsInterval * 1000
+                  ).toString();
+
+                  if (response.data.frames[key]) {
+                    this.updateAnnotations(response.data.frames[key]);
+                  }
+
+                  (videoElement as any).requestVideoFrameCallback(
+                    videoFrameCallback
+                  );
+                };
+
+                if ("requestVideoFrameCallback" in HTMLVideoElement.prototype) {
+                  (videoElement as any).requestVideoFrameCallback(
+                    videoFrameCallback
+                  );
+                }
+              }
+            })
+            .catch(error => {
+              let message = `Failed to predict video. ${error}`;
+              if (error.response) {
+                message = `${error.response.data.error}: ${error.response.data.message}`;
+              }
+
+              CreateGenericToast(message, Intent.DANGER, 3000);
+            });
+        }
+      })
+    );
+
+    await this.updateImage();
+    this.setState({ predictDone: 0, uiState: null });
+  }
+
```

Comment:
Not sure whether you could refactor `bulkAnalysis` and `getInference`. The way I see it, they have very similar code blocks

Reply:
ok will make the change after the cachelist PR is merged!

Reply:
@virtuoso98 ZhaoYuan, I've made the requested changes and have refectored the code such that getInference() is now a general method called by bulkAnalysis() and singleAnalysis()

Reply:
@beesaycheese , thanks for the changes! Looks good to me